metric-anomaly-data-model/src/main/java/org/hypertrace/alert/engine/metric/anomaly/datamodel/queue/KafkaAlertTaskConsumer.java

```diff
@@ -0,0 +1,72 @@
+package org.hypertrace.alert.engine.metric.anomaly.datamodel.queue;
+
+import com.typesafe.config.Config;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Properties;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.hypertrace.alert.engine.metric.anomaly.datamodel.AlertTask;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KafkaAlertTaskConsumer {
+  private static final Logger LOGGER = LoggerFactory.getLogger(KafkaAlertTaskConsumer.class);
+  private static final int CONSUMER_POLL_TIMEOUT_MS = 100;
+
+  private KafkaConfigReader kafkaConfigReader;
+  private KafkaConsumer<String, ByteBuffer> consumer;
+  private LinkedList<ConsumerRecord<String, ByteBuffer>> linkedList = new LinkedList<>();
+
+  public KafkaAlertTaskConsumer(Config kafkaQueueConfig) {
+    this.kafkaConfigReader = new KafkaConfigReader(kafkaQueueConfig);
+    Properties props = new Properties();
+    props.putAll(kafkaConfigReader.getConsumerConfig(createBaseProperties()));
+    consumer = new KafkaConsumer<String, ByteBuffer>(props);
+    consumer.subscribe(Collections.singletonList(kafkaConfigReader.getTopicName()));
+  }
+
+  public AlertTask dequeue() throws IOException {
```

Comment:
we could have removed this method as well and renamed the class to KafkaConsumer

Reply:
Well, this is not a general consumer, for now, it is producing alert task and consuming it. We will convert to queue interface as follow.